flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/SinkMetricsITCase.java

```diff
@@ -130,8 +132,10 @@ private void assertSinkMetrics(
         int subtaskWithMetrics = 0;
         for (OperatorMetricGroup group : groups) {
             Map<String, Metric> metrics = reporter.getMetricsByGroup(group);
-            // there are only 2 splits assigned; so two groups will not update metrics
-            if (group.getIOMetricGroup().getNumRecordsOutCounter().getCount() != 0) {
+            // There are only 2 splits assigned; so two groups will not update metrics.
+            // There is no other way to access the counter via OperatorMetricGroup, we have to use
+            // metrics from the reporter.
+            if (((Counter) metrics.get(MetricNames.NUM_RECORDS_SEND)).getCount() == 0) {
```

Comment:
The test changed from using the records **out** counter to records **send** counter. I guess it should be consistent.

Reply:
Could you explain what does "consistent" mean in this case? The `numRecordsOutCounter` is defined in the `OperatorIOMetricGroup` and the `numRecordsSendCounter` is in the `SinkWriterMetricGroup`. Since they are on different abstraction levels and there is no way to access the `numRecordsSendCounter` from `OperatorMetricGroup`, how could they be consistent?

Reply:
You actually made it consistent.